src/hardware/driver/ltc2320.rs

```diff
@@ -0,0 +1,130 @@
+///! LTC2320 Driver
+///!
+///! QSPI bug (2.4.3):
+///! https://www.st.com/resource/en/errata_sheet/es0392-stm32h742xig-and-stm32h743xig-device-limitations-stmicroelectronics.pdf
+///!
+///! This driver is intended to be used in the following manner:
+///! 1. Trigger a new LTC2320 conversion with start_conversion(). This sets nCNV low and starts
+///!    a hardware timer to trigger en interrupt when TCONV has passed.
+///! 2. Call handle_conv_done_irq() in the timer ISR to stop and reset the timer and start the
+///!    QSPI readout. The QSPI peripheral will trigger another interrupt once that transfer is done.
+///! 3. Call handle_transfer_done_irq() in the QSPI ISR to retrieve the ADC data and set nCNV high again.
+///!
+///! Restarting conversions faster than (T_readout + TCONV + TCNVH) can lead to undefiened behaviour!
+use super::super::hal::{
+    device::QUADSPI,
+    gpio::{self, gpiob, gpioc, gpioe},
+    pac,
+    prelude::*,
+    rcc, stm32,
+    time::NanoSeconds,
+    timer::{self, Timer},
+    xspi::{Qspi, QspiMode, XspiExt},
+};
+use core::ptr;
+
+pub struct Ltc2320Pins {
+    pub spi: (
+        gpiob::PB2<gpio::Alternate<9>>,
+        gpioe::PE7<gpio::Alternate<10>>,
+        gpioe::PE8<gpio::Alternate<10>>,
+        gpioe::PE9<gpio::Alternate<10>>,
+        gpioe::PE10<gpio::Alternate<10>>,
+    ),
+    pub cnv: gpioc::PC11<gpio::Output<gpio::PushPull>>,
+}
+
+pub struct Ltc2320 {
+    qspi: Qspi<QUADSPI>,
+    cnv: gpioc::PC11<gpio::Output<gpio::PushPull>>,
+    timer: Timer<stm32::TIM7>,
+}
+
+impl Ltc2320 {
+    const N_BYTES: usize = 17; // Number of bytes to be transfered. One extra to account for dummy address.
+    const TCONV: u32 = 450; // minimum conversion time according to datasheet
+    pub fn new(
+        clocks: &rcc::CoreClocks,
+        qspi_rec: rcc::rec::Qspi,
+        qspi_peripheral: stm32::QUADSPI,
+        timer_rec: rcc::rec::Tim7,
+        timer_peripheral: stm32::TIM7,
+        mut pins: Ltc2320Pins,
+    ) -> Self {
+        let mut qspi =
+            qspi_peripheral.bank2(pins.spi, 1u32.MHz(), clocks, qspi_rec);
```

Comment:
I just wanted to be relatively slow to look at scope traces. I'd make the decision for the SPI frequency once we have hardware. Since we don't use the return clock it's not so easy to say what the ideal frequency is, right?